src/main/java/com/whytowait/core/annotations/CheckMerchantManagerRole.java

```diff
@@ -0,0 +1,22 @@
+package com.whytowait.core.annotations;
+
+import com.whytowait.core.annotations.enums.RequestSource;
+import com.whytowait.domain.models.enums.MerchantRole;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ElementType.METHOD, ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface CheckMerchantManagerRole {
+
+    MerchantRole requiredAuthority();
+
+    RequestSource source();
+
```

Comment:
Here add the fieldName String type - field and you can also set default value of this field Name.
Change name of requiredAuthority to requiredRole

e.g. usage will be `@CheckMerchantManagerRole(source = RequestSource.PARAM, requiredRole = MerchantRole.MERCHANT_MANAGER, fieldName = "merchId")`